src/api/v1beta1/properties.go

```diff
@@ -487,3 +488,27 @@ func (dk *DynaKube) GetOneAgentEnvironment() []corev1.EnvVar {
 	}
 	return []corev1.EnvVar{}
 }
+
+func (dk *DynaKube) ShallUpdateOneAgentConnectionInfo() bool {
+	return dk.shallUpdateOneAgentConnectionInfo(timeutil.NewProvider())
+}
+
+func (dk *DynaKube) shallUpdateOneAgentConnectionInfo(timeProvider *timeutil.Provider) bool {
+	return timeutil.TimeoutReached(&dk.Status.DynatraceApi.LastOneAgentConnectionInfoRequest, timeProvider.Now(), dk.FeatureApiRequestThreshold())
+}
+
+func (dk *DynaKube) ShallUpdateActiveGateConnectionInfo() bool {
+	return dk.shallUpdateActiveGateConnectionInfo(timeutil.NewProvider())
+}
+
+func (dk *DynaKube) shallUpdateActiveGateConnectionInfo(timeProvider *timeutil.Provider) bool {
+	return timeutil.TimeoutReached(&dk.Status.DynatraceApi.LastActiveGateConnectionInfoRequest, timeProvider.Now(), dk.FeatureApiRequestThreshold())
+}
+
+func (dk *DynaKube) ShallVerifyTokenScope() bool {
+	return dk.shallVerifyTokenScope(timeutil.NewProvider())
+}
+
+func (dk *DynaKube) shallVerifyTokenScope(timeProvider *timeutil.Provider) bool {
+	return timeutil.TimeoutReached(&dk.Status.DynatraceApi.LastTokenScopeRequest, timeProvider.Now(), dk.FeatureApiRequestThreshold())
+}
```

Comment:
the whole point of the `timeProvider` is that it makes time mockable, but you only allow that in for this package, 
so any package that relies on any of the functions can't mock time, which is not the intended outcome. Its really simple to create a `timeProvider` in the reconciler, you could just create it in the reconciler's constructor for now, and we can improve that later if we want to.

furthermore you could have used the `timeProvider.IsOutdated` instead of the `TimeoutReached` as it does the same, but cleaner. (if you are already using a `timeProvider`)

please dont use `Shall...` as a prefix for boolean functions, its really unorthodox and just weird to read 😅 


example:
```
func (dk *DynaKube) IsOutdatedTokenScope(timeProvider *timeutil.Provider) bool {
	return timeProvider.IsOutdated(&dk.Status.DynatraceApi.LastTokenScopeRequest, dk.FeatureApiRequestThreshold())
}
```

Reply:
True, kind of got lost in details here, I guess.